### PR TITLE
Deprecate the Build trait

### PR DESCRIPTION
**Pull request by dwijnand**

### CODE REVIEW

File: main/src/main/scala/sbt/Build.scala [hunk 1]
```diff
@@ -8,6 +8,7 @@ import Keys.{ name, organization, thisProject, autoGeneratedProject }
 import Def.{ ScopedKey, Setting }
 
 // name is more like BuildDefinition, but that is too long
+@deprecated("Use .sbt format instead", "0.13.12")
```
Comment (eed3si9n) [inline main/src/main/scala/sbt/Build.scala:11]: I think we should write a migration guide in http://www.scala-sbt.org/0.13/docs/Full-Def.html and link to it from here.

Comment (dwijnand) [inline main/src/main/scala/sbt/Build.scala:11]: Right, yeah. There and in the notes as well.

I'll try and sketch one out. I'll need a little bit of your time, as your documentation-fu is stronger than mine.

**Comment by eed3si9n** [comment]
I'll merge this for now.

**Comment by dwijnand** [comment]
Forward ported with a breaking change in https://github.com/sbt/sbt/pull/2524

**Pull request merged**